platforms/jvm/scala/src/main/java/org/gradle/api/tasks/scala/ScalaTask.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.scala;
+
+import org.gradle.api.Task;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.tasks.Classpath;
+import org.gradle.internal.instrumentation.api.annotations.ToBeReplacedByLazyProperty;
+
+/**
+ * Common interface for Scala tasks.
+ *
+ * @since 8.10
```

Comment:
Let me know whether these changes still have a chance to make it into `8.10` – if not, I'll have to update this.